maltoolbox/language/languagegraph.py

```diff
@@ -26,44 +26,64 @@
 
 logger = logging.getLogger(__name__)
 
+def get_asset_name_from_full_attack_step(
+        attack_step_full_name: str) -> str:
+    return attack_step_full_name.split(':')[0]
+
+def get_attack_step_name_from_full_attack_step(
+        attack_step_full_name: str) -> str:
+    return attack_step_full_name.split(':')[1]
+
 @dataclass
 class LanguageGraphAsset:
-    name: Optional[str] = None
-    associations: list[LanguageGraphAssociation] = field(default_factory=lambda: [])
-    attack_steps: list[LanguageGraphAttackStep] = field(default_factory=lambda: [])
+    name: str
+    associations: dict[str, LanguageGraphAssociation] = \
+        field(default_factory=lambda: {})
+    attack_steps: dict[str, LanguageGraphAttackStep] = \
+        field(default_factory = lambda: {})
     description: dict = field(default_factory=lambda: {})
```

Comment:
Why call this attribute description when it is `info` fields we are copying from the MAL spec? Currently, there are three names for this information: `info` in the MAL spec, `meta` in the langspec.json and `description` here.

Reply:
This makes sense to me. The `meta` name will have to be there for a bit since it's what the current MAL compiler calls it. We may deprecate later and just have converters or legacy compatibility loaders.

But what is it that we want to call this field? The `meta` name is the worst from my point of view. They all feel a bit generic, but maybe we just want to stick to `info` since that's the current MAL specification?

Reply:
I would go with info so that there is a single name throughout the pipeline for this.